modules/puppetca_token_whitelisting/puppetca_token_whitelisting_plugin.rb

```diff
@@ -0,0 +1,11 @@
+module ::Proxy::PuppetCa::TokenWhitelisting
+  class Plugin < ::Proxy::Provider
+    plugin :puppetca_token_whitelisting, ::Proxy::VERSION
+
+    requires :puppetca, ::Proxy::VERSION
+    default_settings :sign_all => false, :hosts_file => '/tmp/foreman-proxy/hosts.yml'
```

Comment:
I don't like this default because `/tmp` is a shared directory. Another user on the system can create `/tmp/foreman-proxy` prior to startup and have control. That makes it a security risk. We do have `/var/lib/foreman-proxy` in some packages (notably REX/Ansible I think) and perhaps we should standardize this. An alternative is doing it in `/etc/foreman-proxy` but I don't like it because it gives the impression users can/should modify it.

Reply:
We also weren't sure where to put the file. Its not code, it's not config, it does not belong in the puppet dir. I settled for this as it is a temporary file, but am not quite happy, so I'm open to suggestions. You think `/var/lib/foreman-proxy/` would be the best idea?

Reply:
`/tmp` is wiped after a reboot. Losing that kind of state doesn't sound right for this file. `/var/tmp` would be better but to me it sounds like this is actually a state file. For that `/var/lib` is generally used.